core/src/main/java/tc/oc/pgm/listeners/PGMListener.java

```diff
@@ -258,6 +261,30 @@ public void handleItemPickup(final PlayerPickupItemEvent event) {
     if (nearestPlayer != event.getPlayer()) event.setCancelled(true);
   }
 
+  @EventHandler
+  public void lockFireTick(final MatchLoadEvent event) {
+    oldFireTick = Boolean.parseBoolean(event.getMatch().getWorld().getGameRuleValue(DO_FIRE_TICK));
+    event.getMatch().getWorld().setGameRuleValue(DO_FIRE_TICK, Boolean.toString(false));
+  }
+
+  // if no doFireTick <gamerule> is found, then use doFireTick setting saved in world
+  @EventHandler
+  public void unlockFireTick(final MatchStartEvent event) {
+    ImmutableMap<String, String> gameRules =
+        event.getMatch().getModule(GameRulesMatchModule.class).getGameRules();
+    String newFireTick = gameRules.get(DO_FIRE_TICK);
+    if (!gameRules.isEmpty() && newFireTick != null) {
+      event.getMatch().getWorld().setGameRuleValue(DO_FIRE_TICK, newFireTick);
+    } else {
+      event.getMatch().getWorld().setGameRuleValue(DO_FIRE_TICK, Boolean.toString(oldFireTick));
+    }
```

Comment:
don't copy-paste code in 2 diff invocations depending on an if for a boolean parameter, make them part of the parameter passed directly.

```suggestion
    event.getMatch().getWorld().setGameRuleValue(DO_FIRE_TICK, 
            !gameRules.isEmpty() && newFireTick != null ? newFireTick : Boolean.toString(oldFireTick));
```
however, this logic should be even simpler with what i proposed above